cli/cli/main.go

```diff
@@ -40,8 +49,59 @@ func main() {
 	})
 
 	if err := commands.RootCmd.Execute(); err != nil {
-		// We don't actually need to print the error because Cobra will do it for us
+		maybeCleanedError := getErrorMessageToBeDisplayedOnCli(err)
+		errorMessageFromCli := maybeCleanedError.Error()
+		// cobra uses this method underneath as well to print errors
+		// so just used it directly.
+		commands.RootCmd.PrintErrln(errorPrefix, errorMessageFromCli)
+
+		// if unknown command is entered - display help command
+		if strings.Contains(errorMessageFromCli, commandNotFound) {
+			commands.RootCmd.PrintErrf("Run '%v --help' for usage.\n", commands.RootCmd.CommandPath())
+		}
 		os.Exit(errorExitCode)
 	}
 	os.Exit(successExitCode)
 }
+
+func getErrorMessageToBeDisplayedOnCli(errorWithStacktrace error) error {
+	// if we are running in the debug mode, just return the error with stack-traces back to the client
+	if logrus.GetLevel() == logrus.DebugLevel {
+		return errorWithStacktrace
+	}
+
+	// silently catch the file logger error and print it in the debug mode
+	// users should not worry about this error
+	// downside is that we may lose stack-traces during file logger failures
+	fileLogger, err := out.GetFileLogger()
+	if err != nil {
+		logrus.Debugf("Error occurred while getting the file logger %+v", err)
+	} else {
+		fileLogger.Errorln(errorWithStacktrace.Error())
+	}
+
+	errorMessage := errorWithStacktrace.Error()
+	cleanError := removeFilePathFromErrorMessage(errorMessage)
+	return cleanError
+}
+
+// this method removes the file path from the error
+func removeFilePathFromErrorMessage(errorMessage string) error {
+	errorMessageConvertedInList := strings.Split(errorMessage, separator)
+	// safe to assume that the error was not generated using stacktrace package
+	if len(errorMessageConvertedInList) == errorNotCreatedFromStacktrace {
+		return errors.New(errorMessage)
+	}
+
+	// only the even numbered elements needs to be picked.
+	var cleanErrorList []string
+	for index, line := range errorMessageConvertedInList {
+		if index%2 == 0 {
```

Comment:
Oh wow, this is ... hacky (this, why not), but what worries me the most here is that it is also super fragile. Let's saying tomorrow I implement a command where somewhere a function returned to following error:
```
returns stacktrace.Propagate("I am breaking --- the log cleaning --- logic --- hehehehe")
```
Since I introduced and odd number of what you call separator here, that will mess up your entire logic, and post this line you'll end up printing only the path to files, not the "Caused by"

What I'm trying to say here is that in those situations where we feel the behaviour we're introducing is hacky (sometime hacks are the only way, there's no good alternative), we should try to bias toward what's the safest. What would be safer here would be to have a good regexp to match the lines you want to mute.
If the regexp becomes out of date at some point, then at worse we will start printing those undesired lines again, which is not great, but at least we biased toward not hiding relevant info to the users.

Does that make sense? I feel a bit uncomfortable with this TBH, but let me know if there's something I missed

Reply:
discussed offline - will do that in next PR ( will add a regex to make this more strict.